src/__experimental__/components/date/date.component.js

```diff
@@ -191,15 +192,31 @@ class BaseDateInput extends React.Component {
       this.isOpening = false;
       return;
     }
+
     document.removeEventListener('click', this.closeDatePicker);
-    this.setState({ isDatePickerOpen: false }, () => {
+    this.setState({ isDatePickerOpen: false, shouldPickerOpen: false }, () => {
       this.isBlurBlocked = false;
       if (this.input !== document.activeElement) {
         this.handleBlur();
       }
     });
   };
 
+  handleClick = () => {
+    this.setState(_prevState => ({
+      shouldPickerOpen: !_prevState.shouldPickerOpen
+    }), () => {
+      if (!this.state.shouldPickerOpen) {
+        this.inputFocusedViaPicker = false;
+        this.closeDatePicker();
+        console.log('here');
```

Comment:
leftover debug code?

Reply:
Yep. Jamie has pointed it out. I'll get it removed now.